nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -270,33 +273,111 @@ bool CollisionMonitor::getParameters(
       const std::string polygon_type = get_parameter(polygon_name + ".type").as_string();
 
       if (polygon_type == "polygon") {
-        std::shared_ptr<Polygon> p = std::make_shared<Polygon>(
-          node, polygon_name, base_frame_id, simulation_time_step);
-
-        if (!p->configure()) {
-          return false;
-        }
-
-        polygons_.push_back(p);
+        polygons_.push_back(std::make_shared<Polygon>(node, polygon_name));
       } else if (polygon_type == "circle") {
-        std::shared_ptr<Circle> c = std::make_shared<Circle>(
-          node, polygon_name, base_frame_id, simulation_time_step);
-
-        if (!c->configure()) {
-          return false;
-        }
-
-        polygons_.push_back(c);
+        polygons_.push_back(std::make_shared<Circle>(node, polygon_name));
       } else {  // Error if something else
         RCLCPP_ERROR(
           get_logger(),
           "[%s]: Unknown polygon type: %s",
           polygon_name.c_str() ,polygon_type.c_str());
         return false;
       }
+
+      // Configure last added polygon
+      if (!polygons_.back()->configure()) {
+        return false;
+      }
+
+      // Added polygon should be STOP or SLOWDOWN action type
+      if (polygons_.back()->getActionType() != STOP &&
+        polygons_.back()->getActionType() != SLOWDOWN)
+      {
+        RCLCPP_ERROR(
+          get_logger(),
+          "[%s]: Should be \"stop\" or \"slowdown\" action type",
+          polygon_name.c_str());
+        return false;
+      }
     }
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(get_logger(), "Error while getting parameters: %s", ex.what());
+    return false;
+  }
 
-    // Working with data sources parameters
+  return true;
+}
+
+bool CollisionMonitor::configureFootprint(std::string & footprint_topic)
+{
+  try {
+    auto node = shared_from_this();
+
```

Comment:
None of this should be here. This is just a polygon, why would this not be in the same `getParameters()` or `initialize` function as any other polygon type?

Reply:
This was unified (back) in multi-footprint model according to latest update